### DISCUSSION TITLE
Improvements to translations

### DISCUSSION THREAD

**Discussion by Andre601** [QuickShop-Community/QuickShop-Hikari#64] [Ideas]
Not a big fan of those particular strings:
![image](https://user-images.githubusercontent.com/11576465/157500014-26f4ebf1-c707-4f6b-b734-467ba9a53c1c.png)

You should consider using something like a StringBuilder, which would use parts from the translations to construct the final String, because all those lines are either pre-defined text without a hover, or placeholder text with a hover.

In summary could the text be split up into those components:

- `&b|` for separation
- `[{0}](color={1})` for text without hover
- `[{0}](color={1} hover={2})` for text with hover
- The different, pre-defined Strings for the options:
  - `owner`
  - `player`
  - `action`
  - `shop`
  - `price`
  - `purchaser`
  - `amount`
  - `tax`

Then QuickShop would pick either variant and combine them in a StringBuilder before parsing it for final sending.

----

On another note would I like to state my disliking of the MineDown formatting.
I hate this library because the text starts to look extremely ugly with some strange formatting choices at different places.

Personally, am I prefering the MiniMessage formatting. Not only because servers such as PaperMC already support it, but also because it IMO has a bit cleaner look compared to MineDown.
It would also offer a Markdown-like parsing solution (Currently not implemented in the latest adventure because of issues) that would allow customization of different "flavours" like GFM or Discord's Markdown.

**Comment by Ghost-chu** [comment]
will consider add MiniMessage supports :)

**Reply by Ghost-chu** [reply]
https://github.com/Ghost-chu/QuickShop-Hikari/commit/9e091c737aff6fb05734ff8bcdfa2dafedeca9cc

**Reply by Andre601** [reply]
What about the other thing? The one about using a StringBuilder or StringJoiner to construct the final String on the fly?
The lines I showed all share the same parts and the only differences seem to be if they have hover text or not, so you could easily make a function/system in the plugin that constructs the final String using parts from the translations to build.

**Reply by Andre601** [reply]
Another point I mentioned to you on Discord is the strange, mixed use of MineDown and MiniMessage (Or are you using the Markdown feature of MiniMessage?
If you're using both MiniMessage and Minedown should you perhaps fully switch to MiniMessage and instead use the `<hover:show_text='text'>Text</hover>` format for consistency. To my knowledge can you apply the same MiniMessage formatting in the hover text, so `<gold>` could f.e. be used.

**Reply by Ghost-chu** [reply]
> Another point I mentioned to you on Discord is the strange, mixed use of MineDown and MiniMessage (Or are you using the Markdown feature of MiniMessage? If you're using both MiniMessage and Minedown should you perhaps fully switch to MiniMessage and instead use the format for consistency. To my knowledge can you apply the same MiniMessage formatting in the hover text, so could f.e. be used.`<hover:show_text='text'>Text</hover>``<gold>`

minedown has been removed, old minedown strings still unused and will update once I complet the new features.

That is a good idea, but there has a big issue -- all versions of hikari using same file on crowdin, I need thinking about backward compatibility.

But I will definitely use them in new features strings.
